c/bpxskt.c

```diff
@@ -781,7 +783,9 @@ int tcpIOControl(Socket *socket, int command, int argumentLength, char *argument
           returnCode,
           reasonCodePtr);
   if (returnValue < 0){
-    printf("ioctl failed ret code %d reason 0x%x\n",*returnCode,*reasonCode);
+    if (socketTrace){
+      printf("ioctl failed ret code %d reason 0x%x\n",*returnCode,*reasonCode);
```

Comment:
This just may not be needed at all, because you will know success by getting a return of NULL.